### ISSUE TITLE
Broken `ncu -g` with Error since v16.14.13

### ISSUE THREAD

**Issue by ath0mas** [raineorshine/npm-check-updates#1365]
- [x] I have searched for [similar issues](https://github.com/raineorshine/npm-check-updates/issues)
- [x] I am using the latest version of `npm-check-updates`
- [x] I am using `node >= 14.14`

Node 21.6.1
Npm 10.2.4

---

## Steps to Reproduce

.ncurc:

```js

```

Dependencies:

```json

```

Steps:

`ncu -g`

## Current Behavior

```
Error: Expected JSON from "npm.cmd ls --json --location=global". Instead received: Unknown command: "-"

To see a list of supported npm commands, run:
  npm help

    at parseJson (C:\Users\myuser\scoop\persist\nodejs\bin\node_modules\npm-check-updates\build\src\package-managers\npm.js:246:15)
    at Object.list (C:\Users\myuser\scoop\persist\nodejs\bin\node_modules\npm-check-updates\build\src\package-managers\npm.js:569:26)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async getInstalledPackages (C:\Users\myuser\scoop\persist\nodejs\bin\node_modules\npm-check-updates\build\src\lib\getInstalledPackages.js:22:22)
    at async runGlobal (C:\Users\myuser\scoop\persist\nodejs\bin\node_modules\npm-check-updates\build\src\lib\runGlobal.js:17:28)
    at async runUpgrades (C:\Users\myuser\scoop\persist\nodejs\bin\node_modules\npm-check-updates\build\src\index.js:197:26)
```

## Expected Behavior

A "hotfix" consists in installing back v16.14.12:
`npm -g install npm-check-updates@16.14.12`

then `ncu -g` again

```
[====================] 8/8 100%

 npm-check-updates  16.14.12  →  16.14.13

ncu itself cannot upgrade global packages. Run the following to upgrade all global packages:

npm -g install npm-check-updates@16.14.13
```

**Comment by raineorshine** [comment]
Thanks, I'll take a look.

**Comment by vvyman** [comment]
node version v16.17.1 has a same error.
```
Error: Expected JSON from "npm ls --json --location=global". Instead received: Unknown command: "-"

To see a list of supported npm commands, run:
  npm help

    at parseJson (/opt/homebrew/lib/node_modules/npm-check-updates/build/src/package-managers/npm.js:246:15)
    at Object.list (/opt/homebrew/lib/node_modules/npm-check-updates/build/src/package-managers/npm.js:569:26)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
    at async getInstalledPackages (/opt/homebrew/lib/node_modules/npm-check-updates/build/src/lib/getInstalledPackages.js:22:22)
    at async runGlobal (/opt/homebrew/lib/node_modules/npm-check-updates/build/src/lib/runGlobal.js:17:28)
    at async runUpgrades (/opt/homebrew/lib/node_modules/npm-check-updates/build/src/index.js:197:26)
```
Hope to give one more reference.😁

**Comment by EthanTuning** [comment]
Having this same issue as well. Not sure what would help, but I had the same output as the above is showing and I am on Mac OS Sonoma 14.2.1.

**Comment by solrevdev** [comment]
same here: 

```bash
 ncu -g
Error: Expected JSON from "npm ls --json --location=global". Instead received: Unknown command: "-"

To see a list of supported npm commands, run:
  npm help

    at parseJson (/opt/homebrew/lib/node_modules/npm-check-updates/build/src/package-managers/npm.js:246:15)
    at Object.list (/opt/homebrew/lib/node_modules/npm-check-updates/build/src/package-managers/npm.js:569:26)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async getInstalledPackages (/opt/homebrew/lib/node_modules/npm-check-updates/build/src/lib/getInstalledPackages.js:22:22)
    at async runGlobal (/opt/homebrew/lib/node_modules/npm-check-updates/build/src/lib/runGlobal.js:17:28)
    at async runUpgrades (/opt/homebrew/lib/node_modules/npm-check-updates/build/src/index.js:197:26)

---    
ncu --version
16.14.13
---
node -v
v21.6.1

npm -v
10.4.0

computer_info
----------------------------
OS: macOS 14.3 23D56 arm64
Host: Macmini9,1
Kernel: 23.3.0
Uptime: 1 day, 15 hours, 36 mins
Packages: 380 (brew)
Shell: zsh 5.9
Resolution: 2560x1440, 2560x1440
DE: Aqua
WM: Quartz Compositor
WM Theme: Blue (Light)
Terminal: iTerm2
Terminal Font: CascadiaMonoPL-Regular 16
CPU: Apple M1
GPU: Apple M1
Memory: 2205MiB / 16384MiB
```

**Comment by EriksRemess** [comment]
I think this was fixed in https://github.com/raineorshine/npm-check-updates/commit/50ec0b23db3800c6f97aead24e3ca11d940156a4#diff-97685716b4ca526530febb50f52d242ace881b7e1e77335325f50a554419423dL523

**Comment by yhx-12243** [comment]
> I think this was fixed in https://github.com/raineorshine/npm-check-updates/commit/50ec0b23db3800c6f97aead24e3ca11d940156a4#diff-97685716b4ca526530febb50f52d242ace881b7e1e77335325f50a554419423dL523

This commit is so old that is not the cause of the error. See my comment on the corresponding commit in this issue board.

**Comment by rafrafek** [comment]
May be a good idea to add `ncu -g` to the automated tests.

**Comment by raineorshine** [comment]
Fixed and published in `v16.14.14`. Thanks everyeone.

`ncu -g` was ostensibly under test coverage, but errors in global mode were returning the wrong exit code, which was causing a test false positive. I fixed this in a86905b and added more specific testing of the output, so this should not happen again.

**Comment by ath0mas** [comment]
> Fixed and published in `v16.14.14`. Thanks everyeone.
> 
> `ncu -g` was ostensibly under test coverage, but errors in global mode were returning the wrong exit code, which was causing a test false positive. I fixed this in a86905b and added more specific testing of the output, so this should not happen again.

Thanks @raineorshine 
I won't be able to check this quickly, but I'm sure it's OK, and people around will tell 😉
